04week/mastermind.js

```diff
@@ -28,15 +30,82 @@ function getRandomInt(min, max) {
   return Math.floor(Math.random() * (max - min)) + min;
 }
 
-function generateHint() {
-  // your code here
+function generateHint(guess) {
+
+  /*
+  * This function checks for matches between the guess and solution
+  * Returns the matching letters (in Red) or letter & positions (in Green)
+  *
+  * @param: guess - the guess entered into the prompt
+  *
+  * @return: Color coded numbers
+  */
+
+  let guessArray = guess.split('');
+  let solutionArray = solution.split('');
+
+// Find correct letter locations
+  let correctLocation = 0;
+  let j = 0;
+
+  while(j < guessArray.length) {
+    if (solutionArray[j] === guessArray[j]){
+      correctLocation = correctLocation + 1;
```

Comment:
if you do not set the correctLocation[j] to null, then it will get counted again when you do the next loop to get the correct letters count